reframe/core/pipeline.py

```diff
@@ -799,7 +749,13 @@ def __new__(cls, *args, _rfm_use_params=False, **kwargs):
                         os.path.dirname(inspect.getfile(cls))
                     )
 
-        obj._rfm_init(name, prefix)
+        # Apply the hooks only once (not when deep-copying)
+        if not hasattr(cls, '_rfm_first_init'):
+            cls._rfm_pipeline_hooks.apply(obj)
+            cls._rfm_first_init = True
+
```

Comment:
What I'm missing here is some kind of health check that makes sure that all the hooks are properly attached to a legit stage of the pipeline. I think this could be done by passing here a list with the supported phases that a hook can live on into a sanity function from the `HookRegistry`. Then, in order to not have redundant executions of this sanity function, we could use the `_rfm_use_params` variable, which is only set to `True` by the `simple_test` decorator (it would not run when deep-copying). Of course, if we go for that option, it'd be sensible to rename `_rfm_use_params` to something like `_rfm_registered`.